### ISSUE TITLE
Camera component stops working on Android after a view is rendered on top of it. ANDROID- Expo 32

### ISSUE THREAD

**Issue by otaviogaiao** [expo/expo#3495]
### Environment

  Expo CLI 2.10.1 environment info:
    System:
      OS: macOS High Sierra 10.13.6
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 9.10.0 - ~/.nvm/versions/node/v9.10.0/bin/node
      Yarn: 1.9.4 - ~/.yarn/bin/yarn
      npm: 5.6.0 - ~/.nvm/versions/node/v9.10.0/bin/npm
    IDEs:
      Android Studio: 3.1 AI-173.4819257
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^32.0.0 => 32.0.0 
      react: 16.5.0 => 16.5.0 
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz => 0.57.1 
      react-navigation: ^3.0.9 => 3.0.9 
    npmGlobalPackages:
      expo-cli: 2.10.1

### Steps to Reproduce

1. Open a screen with a Camera component from Expo.
2. Click a button to show a View on Top of the camera.
3. The Camera stops working.

### Expected Behavior

If I render a view on top of the Camera component (actually as its child), the camera should keep working.

### Actual Behavior

On Android, if I have the camera component on a screen, and then I render a view on top of it, the camera stops working. It just shows a black screen. Even if I close the View, the camera stays broken. It only comes back to life if I unmount the screen and mount again (using the goBack function with react navigation in my case).

**NOTE THAT THIS HAPPENS ONLY ON ANDROID**

on iOS, it works flawless.
(Write what happened. Add screenshots!)

### Reproducible Demo

https://snack.expo.io/@otaviogaiao/camera-error-expo

Obviously you need an Android phone to reproduce the issue.

I used a Xiaomi redmi note 3, with Android 6.

EDIT: just tried on a Samsung Galaxy S9+ with Android 9 and the same problem occurred.

**Comment by otaviogaiao** [comment]
Another thing, I found out that once I use the pausePreview method, the camera will break. resumePreview doesn't work on Android. iOS is fine.

**Comment by mmmguitar** [comment]
I'm a little unsure of the full extent of this but we've been having similar issues.  Long story short, seems adding any real components like <Text> over the camera break it, but just having views + overlays over it seems ok.  This appears to be broken on expo 29 as well in a way that the camera appears to work, but invoking camera.takePictureAsync(...) will hang, i.e. callback with the picture never comes.

Take a look at this snack https://snack.expo.io/Bkno8gx_E, I've hacked the example from the bug to have just a View component.  On my phone the camera still works with a white square in it.

`<Camera style={{flex:1}}>
          <View style={{width: '50%', height: 200, backgroundColor: '#ffffff', justifyContent: 'space-between'}} />
      </Camera>`

Long story.  We're on expo 29 and use either a view and/or an SVG overlay to guide picture taking.  With android for some reason it takes 2-4 seconds for it to save the picture with the preview still running, so we hack a dark overlay with some take say "We're saving your image" to smoothe over the UI experience.

This always worked fine until it seems Android 9.0 now has been released. Now we're getting users with the app complaining about the image taking hanging on the "saving" bit.  Seems Android 8 users are fine. 

The workaround for us has been on the take picture we turn the overlay view black to get rid of the preview and make it look like the picture was taken.  We then put the UI text "We're saving your image" below the camera (the camera is in a frame / not full screen).

Seems quite related to this issue in the way components rendered over the camera break it but simple overlays appear to work.  Also that Android 9 seems to break this more, Android 8.0 and below worked ok.
